pkg/cmd/experimental/buildchain/buildchain.go

```diff
@@ -94,12 +94,20 @@ func (o *BuildChainOptions) Complete(f *clientcmd.Factory, cmd *cobra.Command, a
 	}
 	o.c, o.t = oc, oc
 
-	// Parse user input
-	o.name, o.tag, err = buildChainInput(args[0])
+	kind, defaultKind := "", "ImageStreamTag"
+	mapper, _ := f.Object()
+	kind, o.name, err = osutil.ResolveResource(defaultKind, args[0], mapper)
```

Comment:
I think what you really want is a way to get a canonical resource representation.  Something where you get "imagestreamtag" back for "istag", "istags", "imagestreamtags" in any case.  If multiple resources deal in the same kind, this breaks because "notAnIsTag/foo" would match.

I asked @smarterclayton if he knew whether it existed, he said he'd look for it/make it.